### PR TITLE
Add unit test for clientConfig.go

### PR DESCRIPTION
**Pull request by vchepeli**
**Describe what this PR does and why we need it**:

Changes proposed in this pull request
 -
 -
 -

**Does this PR depend on another PR (Use this to track when PRs should be merged)**
depends-on <PR>

**Which issue this PR fixes (This will close that issue when PR gets merged)**
fixes <issue_number>

### CODE REVIEW

File: pkg/cmd/clientConfig_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+import (
+	"reflect"
+	"testing"
+
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	kFake "k8s.io/client-go/kubernetes/fake"
+)
+
+type args struct {
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: Not sure on this, why have a type like this in the test?

File: pkg/cmd/clientConfig_test.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	k8Client kubernetes.Interface
+}
+
+func TestNewClientConfigCmd(t *testing.T) {
+	k8Client := &kFake.Clientset{}
+	tests := []struct {
+		name string
+		args args
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: I think we could just replace this with the kubernetes.Interface or a function that returns kubernetes.Interface. I assume the idea with the args here is that is can be expanded upon in the future?
Also while nothing is manipulating the k8client, it is a pointer and so if the test where extended in the future, you might endup with test order problems where one test is changing the behaviour of the client and the next test is using the same client. Overall I prefer to keep the behaviour locked down to the individual test case. I have found having a func 
``` func()kubernetes.Interface{}``` to be the best approach overall

File: pkg/cmd/clientConfig_test.go [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			ccCmd := &ClientConfigCmd{
+				k8Client: tc.args.k8Client,
+			}
+
+			got := ccCmd.GetClientConfigCmd()
+			if use := got.Use; !reflect.DeepEqual(use, fakeCbrCmd.Use) {
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: Use is just a string here so no need for reflection, just us ``` if got.User != fakeCbrCmd.Use``` same for the others below

File: pkg/cmd/clientConfig_test.go [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+				t.Errorf("ClientConfigCmd.GetClientConfigCmd().Use = %v, want %v", use, fakeCbrCmd.Use)
+			}
+			if short := got.Short; !reflect.DeepEqual(short, fakeCbrCmd.Short) {
+				t.Errorf("ClientConfigCmd.GetClientConfigCmd().Short = %v, want %v", short, fakeCbrCmd.Short)
+			}
+			if runE := got.RunE; !reflect.DeepEqual(reflect.TypeOf(runE), reflect.TypeOf(fakeCbrCmd.RunE)) {
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: Not sure this could ever fail so don't think we need this check

File: pkg/cmd/clientConfig_test.go [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+				t.Errorf("ClientConfigCmd.GetClientConfigCmd().RunE = %v, want %v", reflect.TypeOf(runE), reflect.TypeOf(fakeCbrCmd.RunE))
+			}
+			runE := got.RunE
+			fakeCbrCmd.Flags().String("namespace", "", "Namespace for software installation")
+			err := runE(fakeCbrCmd, nil) // args are not used in RunE function
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: Could we change this test to work more like the one here:
https://github.com/aerogear/mobile-cli/blob/master/pkg/cmd/clients_test.go#L77
Notice the ```ExpectError ``` if you need any help with the change let me know

File: pkg/cmd/clientConfig_test.go (hunk 1 above)
Comment (vchepeli) [reply pkg/cmd/clientConfig_test.go]: https://github.com/aerogear/mobile-cli/blob/master/pkg/cmd/clientConfig.go#L33, but agree that I have only one instance of of same client. Need change to function

File: pkg/cmd/clientConfig_test.go [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+		{
+			name:     "new client config command",
+			k8Client: fakeK8Client,
+			want: &ClientConfigCmd{
+				k8Client: fakeK8Client,
+			},
+		},
+	}
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
```
Comment (maleck13) [inline pkg/cmd/clientConfig_test.go]: I am not entirely sure what this test is testing. I know it is only a small test but struggling to see the benefit as it seems like the compiler should ensure this is the case

Comment (vchepeli) [reply pkg/cmd/clientConfig_test.go]: This test is testing that NewClientConfigCmd returns fake client :) But you are right thats constructor testing. Does not mean any sense

File: pkg/cmd/clients.go [hunk 7]
```diff
@@ -121,7 +129,11 @@ When used standalone, a namespace must be specified by providing the --namespace
 			name := args[0]
 			clientType := args[1]
 			appKey := uuid.NewV4().String()
-			namespace := currentNamespace(cmd.Flags())
+
+			namespace, err := currentNamespace(cmd.Flags())
+			if err != nil {
+				return errors.Wrap(err, "failed to get namespace")
```
Comment (maleck13) [inline pkg/cmd/clients.go]: I am a fan of using this package for errors so gets a 👍  from me

**Pull request merged**